### PR TITLE
chore: run unit tests with asan

### PR DESCRIPTION
**Pull request by messense**

### CODE REVIEW

**Comment by messense** [comment]
Here's an example: https://github.com/messense/napi-rs/runs/4837807932?check_suite_focus=true

```rust
#[napi]
fn contains(source: String, target: String) -> bool {
  let xs = [0, 1, 2, 3];
  let _y = unsafe { *xs.as_ptr().offset(4) };
  source.contains(&target)
}
```

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -79,6 +79,25 @@ jobs:
...
+          profile: minimal
+          override: true
+
+      - name: Unit tests with address sanitizer
+        if: matrix.os == 'ubuntu-latest'
+        run: |
+          yarn build:test
+          LD_PRELOAD=/usr/lib/gcc/x86_64-linux-gnu/9/libasan.so yarn test
+        env:
+          RUST_TARGET: x86_64-unknown-linux-gnu
```
Comment (messense) [inline .github/workflows/test.yaml]: Maybe cli should use the `CARGO_BUILD_TARGET` env var instead.

https://doc.rust-lang.org/cargo/reference/environment-variables.html#configuration-environment-variables

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -79,6 +79,25 @@ jobs:
...
+          RUST_BACKTRACE: 1
+          RUSTFLAGS: -Z sanitizer=address
+          ASAN_OPTIONS: detect_leaks=0
```
Comment (messense) [inline .github/workflows/test.yaml]: Too many false positives with leak sanitizer turned on.

**Comment by Brooooooklyn** [comment]
Wow, great!

**Comment by messense** [comment]
Re-run to see if the issue persists.

**Review by Brooooooklyn — APPROVED** [review]

**Pull request merged**